src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -685,9 +686,11 @@ public Thread newThread(Runnable r) {
     protected LoginService configureUserRealm() {
         HashLoginService realm = new HashLoginService();
         realm.setName("default");   // this is the magic realm name to make it effective on everywhere
-        realm.update("alice", new Password("alice"), new String[]{"user","female"});
-        realm.update("bob", new Password("bob"), new String[]{"user","male"});
-        realm.update("charlie", new Password("charlie"), new String[]{"user","male"});
+        UserStore userStore = new UserStore();
```

Comment:
Forced by Jetty changes as detailed in https://github.com/eclipse/jetty.project/pull/1488#issuecomment-332201285. This makes it awkward to switch Jetty versions in a plugin back to 9.2.x, which @vivek says is currently necessary for Wiremock compatibility: you must use

```java
@Rule
public JenkinsRule r = new JenkinsRule() {
    @Override
    protected LoginService configureUserRealm() {
        HashLoginService realm = new HashLoginService();
        realm.setName("default");
        return realm;
    }
};
```

We cannot simply delete the `UserStore` part from the default implementation; it works for most tests, but not those calling `WebClient.login` while using `LegacySecurityRealm` (perhaps indirectly via `@PresetData(DataSet.NO_ANONYMOUS_READACCESS)`), of which there are a few in core and some plugins. (For example, `JnlpAccessWithSecuredHudsonTest`, using `HudsonTestCase`.)